src/tests/bignum_test.nr

```diff
@@ -1044,3 +1049,132 @@ fn fuzz_cmp_equal(seed: [u8; 5]) {
+    // Test single limb (64-bit number)
+    let x = 0x8000000000000000;
+    let arr = [0, 0, x as u128, 0];
+    let msb1 = unsafe { __get_msb64(arr) };
+    let msb2 = unsafe { __get_msb(arr) };
+    assert_eq(msb1, msb2);
+
+    // Test multiple limbs (120-bit number)
+    let x = 0x800000000000000000000000000000; // 120 bits number
+    let arr = [0, 0, x as u128, 0];
+    let msb1 = unsafe { __get_msb64(arr) };
+    let msb2 = unsafe { __get_msb(arr) };
+    assert_eq(msb1, msb2);
+
+    // Test zero
+    let arr = [0, 0, 0, 0];
+    let msb1 = unsafe { __get_msb64(arr) };
+    let msb2 = unsafe { __get_msb(arr) };
+    assert_eq(msb1, msb2);
+
+    // Test all bits set (120 bits)
+    let x = 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF; // 120 bits
+    let arr = [0, x as u128, 0, 0];
+    let msb1 = unsafe { __get_msb64(arr) };
+    let msb2 = unsafe { __get_msb(arr) };
+    assert_eq(msb1, msb2);
+
+    // Test systematic bit positions
+    for i in 0..120 {
+        let x: u128 = 1;
+        let shifted = x << i;
+        let arr = [0, shifted, 0, 0];
+        let msb1 = unsafe { __get_msb64(arr) };
+        let msb2 = unsafe { __get_msb(arr) };
+        assert_eq(msb1, msb2);
+    }
+
+    // Test random-like patterns (multiple bits set)
+    let patterns = [
+        0xAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA, // alternating bits
+        0x555555555555555555555555555555, // alternating bits (opposite)
+        0x1234567890ABCDEF1234567890ABCD, // some pattern
+        0xFEDCBA0987654321FEDCBA09876543, // some pattern
+        0x800000000000000000000000000001, // highest and lowest bits
+        0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFE, // all bits except lowest
+        0x7FFFFFFFFFFFFFFFFFFFFFFFFFFFFF, // all bits except highest
+    ];
+
+    for pattern in patterns {
+        let arr = [0, pattern as u128, 0, 0];
+        let msb1 = unsafe { __get_msb64(arr) };
+        let msb2 = unsafe { __get_msb(arr) };
+        assert_eq(msb1, msb2);
+    }
+
+    // Test with MSB in different array positions (120 bits)
+    let x = 0x800000000000000000000000000000; // 120 bits
+    let arr1 = [x as u128, 0, 0, 0];
+    let arr2 = [0, x as u128, 0, 0];
+    let arr3 = [0, 0, x as u128, 0];
+    let arr4 = [0, 0, 0, x as u128];
+    let msb1_1 = unsafe { __get_msb64(arr1) };
+    let msb2_1 = unsafe { __get_msb(arr1) };
+    assert_eq(msb1_1, msb2_1);
+
+    let msb1_2 = unsafe { __get_msb64(arr2) };
+    let msb2_2 = unsafe { __get_msb(arr2) };
+    assert_eq(msb1_2, msb2_2);
+
+    let msb1_3 = unsafe { __get_msb64(arr3) };
+    let msb2_3 = unsafe { __get_msb(arr3) };
+    assert_eq(msb1_3, msb2_3);
+
+    let msb1_4 = unsafe { __get_msb64(arr4) };
+    let msb2_4 = unsafe { __get_msb(arr4) };
+    assert_eq(msb1_4, msb2_4);
+}
+
+/*
+#[test]
+unconstrained fn fuzz_get_msb(seed: [u128; 5]) {
+    let mut should_test = true;
+    for i in 0..5 {
+        if (seed[i] as u128) >= TWO_POW_120 {
+            should_test = false;
+            break;
+        }
+    }
```

Comment:
I'd expect this to reject >99% of test cases. We only test if the upper 8 bits of each limb are all set to zero and there's 5 limbs so we get `(1/2)^(8*5),` so we'll only test `~1^(-12)` of cases.

When fuzzing it's preferable to try and massage the input into a valid form rather than just rejecting it outright. We could do this with a bitwise AND with `TWO_POW_120 - 1`